docs/spec.md

```diff
@@ -3690,19 +3715,19 @@ More formally, `result[i0, ..., iR-1]` is equal to:
 
 #### Inputs
 
-| Label | Name                | Type                                         | Constraints      |
-|-------|---------------------|----------------------------------------------|------------------|
-| (I1)  | `operand`           | tensor                                       | (C1), (C2), (C4) |
-| (I2)  | `padding_value`     | 0-dimensional tensor                         | (C1)             |
-| (I3)  | `edge_padding_low`  | 1-dimensional tensor constant of type `si64` | (C2), (C4)       |
-| (I4)  | `edge_padding_high` | 1-dimensional tensor constant of type `si64` | (C2), (C4)       |
-| (I5)  | `interior_padding`  | 1-dimensional tensor constant of type `si64` | (C2-C4)          |
+| Label | Name                | Type                                         | Constraints   |
+|-------|---------------------|----------------------------------------------|---------------|
+| (I1)  | `operand`           | tensor or quantized tensor                   | (C1-C2), (C4) |
+| (I2)  | `padding_value`     | 0-dimensional tensor or quantized tensor     | (C1)          |
```

Comment:
Sorry this is a bit pedantic, but Is padding_value a quantized tensor, or the zero_point of the quantized tensor? In Convolution, you are passing zero_point to pad(), but here it seems to imply you would send a tensor with single value 0 and a zero_point. 
I suppose the above comment where the value used is padding_value[] implies that the zero_point of a passed in padding_value is ignored.

Reply:
Here is my understanding of `stablehlo.pad` op:

`stablehlo.pad` can pad whatever value specified in `pad_value`.  For example, I can decide to pad `5.0` at the boundary of the tensor.

The comment in the `stablehlo.conv` op is re-using the semantics of `stablehlo.pad` but specifies the zero point of the `stablehlo.conv` `input` operand as the pad value.

For example, if the input tensor is `tensor<4x4x!quant.uniform<ui8:f32, 1.0:5>>`, then one step of `stablehlo.conv` will call `stablehlo.pad(value = dense<5> : tensor<ui8:f32, 1.0:5>>)`, where `5` is the zero point.

Maybe need some clarification in the comment in `stablehlo.conv`.

Reply:
Hi @eric-k256 , Thanks for bringing this up. I remember having the same concern while speccing this and tried to address it.

Thanks @loganchien for  capturing the intention here.

I guess part of the confusion  here is that in the convolution op spec, when 'lhs' is a quantized tensor,  we are using the spec of pad op as

```
`padded_lhs_value = is_quantized_tensor(lhs) ? zero_point(lhs) : 0`.
`padded_lhs = pad(lhs, padded_lhs_value, lhs_padding[:, 0], lhs_padding[:, 1], lhs_base_dilations[:] - 1)`.
```

But at this point it is not clear that the `padded_lhs_value` is  a quantized tensor with same element type as `lhs`.  

This is what I had in mind when I wrote this:

1. The fact that the `padded_lhs_value` is of the same element type as `lhs` can be derived from the constraint `C1` of the pad op

```
 (C1) `operand`, `padding_value`, `result` have the same element type.
```

2. I note here that even for the non quantized scenario, when we have
```
`padded_lhs_value =  0`. // as is_quantized_tensor(lhs) is false
`padded_lhs = pad(lhs, padded_lhs_value, lhs_padding[:, 0], lhs_padding[:, 1], lhs_base_dilations[:] - 1)`.
```

We are just passing the just the (zero) value to pad irrespective of the type of `lhs`. This `0` value is expected to be interpreted as a value of type equal to the `lhs`, which could be integer, floating-point, or complex type. 

>> I suppose the above comment where the value used is padding_value[] implies that the zero_point of a passed in padding_value is ignored.

That is true. From the POV of padding operation, specification should just consider the value ignoring the zero_point of the padding value. The zero_point here will just match with the zero_point of the  `operand` per `C1`.

@eric-k256 Please let me know if it makes sense.

Reply:
Yes, this makes sense. Both you and Logan confirmed what I had assumed was the case in this instance.